homework

Zhenlong-Zhang opened the pull request.

Zhenlong-Zhang committed:
homeworkhomework

Wenxuan-Lu commented:
It's a good practice to make smaller, incremental commits as you work, rather than a single large commit for the entire assignment. Consider committing after completing each meaningful part of your work.

Wenxuan-Lu commented on R/utils.R:
-1 pt: These functions are only needed for testing. Since it is not part of the main package functionality, Please place them in the tests/testthat/ folder rather than the R/ folder.

Wenxuan-Lu commented on R/utils.R:
Nice job writing detailed comments for each function. However, not all functions need to be visible from the user side. Please export only functions that are required for the package usage.

Wenxuan-Lu commented on tests/testthat/test-utils.R:
-1pt; Avoid code repetitions. You can use are_all_close() function directly without redefining it here. The same holds for log_likelihood() and grad() functions used to test the gradient calculation.

Wenxuan-Lu commented on tests/testthat/test-utils.R:
In testthat, it is recommended to have one test case per function. Instead of separating test cases with comments, please organize them into separate test_that() function calls.

Wenxuan-Lu commented on tests/testthat/test-gradient.R:
Please write comments in English only to ensure readability for everyone.

Wenxuan-Lu commented on R/hiper_glm.R:
It is recommended to use the function header hiper_glm <- function(design, outcome, model = "linear", option = list()) as shown in Aki's repository. This will make future development easier when more functionality needs to be added.

Wenxuan-Lu commented on R/hiper_glm.R:
The default parameter needs to be a valid input. Please set the default value for "method" to either option but not both.

Wenxuan-Lu commented on R/hiper_glm.R:
-1pt; Since you are calculating negative of log likelihood, it would be better to name the function as neg_log_likelihood to avoid confusion. The same holds for the gradient.

Wenxuan-Lu commented on R/hiper_glm.R:
Great job defining functions for calculating the log-likelihood and gradient log-likelihood. Please move them as functions to a separate file, such as R/likelihood_calculation.R, to keep the code organized. This will make it easier to expand functionalities in the coming weeks.

Zhenlong-Zhang replied:
moved nll and neg grad to a file called likelihood_calculation.R

Wenxuan-Lu commented on R/hiper_glm.R:
Though it does not seem to be necessary for this implementation, moving the actual calculation steps (i.e., lines 11 and 23-25) to separate functions in a separate file would be helpful. This will make the code more modular and flexible, especially for developing a generic function that can accommodate various models and fitting methods.

Zhenlong-Zhang replied:
moved methods to a different file

Wenxuan-Lu commented on R/hiper_glm.R:
-1pt; Please perform a sanity check on the input parameters and terminate the program if the input parameter is invalid.

Wenxuan-Lu commented on R/hiper_glm.R:
```diff
@@ -1,3 +1,29 @@
+
+    beta_init <- solve(t(X) %*% X, t(X) %*% y)  
+    optim_res <- optim(beta_init, log_likelihood, grad, X = X, y = y, method = "BFGS", control = list(fnscale = 1))
+    beta_hat <- optim_res$par
```
-1 pt; Please check whether BFGS has converged before returning the result. If it has not converged, provide a warning message to alert the user.

Wenxuan-Lu reviewed (changes requested):
Great job! I enjoyed reviewing your code. Remember that you can earn back all points by addressing these comments. Please address both point-deducting and non-point-deducting comments. I will update this comment with the final score when HW3 is due.

Zhenlong-Zhang committed:
change R/utils.R to tests/testthat/utils.R

Zhenlong-Zhang committed:
Update test-gradient.R

Zhenlong-Zhang committed:
loglikehood and grad to neg loglikehood and neg grad

Zhenlong-Zhang committed:
log like and grad to neg

Zhenlong-Zhang committed:
BFGS has converged or not, if not warning printed

Zhenlong-Zhang committed:
not showing the users all the functions

not all functions need to be visible from the user side, so export only functions that are required for the package usage

Zhenlong-Zhang committed:
code repetitions

use are all close without redefining

Zhenlong-Zhang committed:
same with last

Zhenlong-Zhang committed:
code repetitions for nll and grad

Zhenlong-Zhang committed:
final version of editing including taking the method out in seprete function,and run the test

Zhenlong-Zhang committed:
one test case per function
